tests/attacks/test_projected_gradient_descent.py

```diff
@@ -294,8 +291,240 @@ def test_scikitlearn(self):
+    @unittest.skipIf(tf.__version__[0] != '2', '')
+    def test_framework_tensorflow_v2_mnist(self):
+        classifier, _ = get_image_classifier_tf()
+
+        scores = get_labels_np_array(classifier.predict(self.x_train_mnist))
+        acc = np.sum(np.argmax(scores, axis=1) == np.argmax(self.y_train_mnist, axis=1)) / self.y_train_mnist.shape[0]
+        logger.info("[TFV2, MNIST] Accuracy on training set: %.2f%%", acc * 100)
+
+        scores = get_labels_np_array(classifier.predict(self.x_test_mnist))
+        acc = np.sum(np.argmax(scores, axis=1) == np.argmax(self.y_test_mnist, axis=1)) / self.y_test_mnist.shape[0]
+        logger.info("[TFV2, MNIST] Accuracy on test set: %.2f%%", acc * 100)
+
+        self._test_framework_vs_numpy(classifier)
+
+    def test_framework_pytorch_mnist(self):
+        self.x_train_mnist = np.swapaxes(self.x_train_mnist, 1, 3).astype(np.float32)
+        self.x_test_mnist = np.swapaxes(self.x_test_mnist, 1, 3).astype(np.float32)
+        classifier = get_image_classifier_pt()
+
+        scores = get_labels_np_array(classifier.predict(self.x_train_mnist))
+        acc = np.sum(np.argmax(scores, axis=1) == np.argmax(self.y_train_mnist, axis=1)) / self.y_train_mnist.shape[0]
+        logger.info("[PyTorch, MNIST] Accuracy on training set: %.2f%%", acc * 100)
+
+        scores = get_labels_np_array(classifier.predict(self.x_test_mnist))
+        acc = np.sum(np.argmax(scores, axis=1) == np.argmax(self.y_test_mnist, axis=1)) / self.y_test_mnist.shape[0]
+        logger.info("[PyTorch, MNIST] Accuracy on test set: %.2f%%", acc * 100)
+
+        self._test_framework_vs_numpy(classifier)
+
+        self.x_train_mnist = np.swapaxes(self.x_train_mnist, 1, 3).astype(np.float32)
+        self.x_test_mnist = np.swapaxes(self.x_test_mnist, 1, 3).astype(np.float32)
+
+    def _test_framework_vs_numpy(self, classifier):
+        # Test PGD with np.inf norm
+        attack_np = ProjectedGradientDescentNumpy(
+            classifier,
+            eps=1,
+            eps_step=0.1,
+            max_iter=5,
+            norm=np.inf,
+            targeted=False,
+            num_random_init=0,
+            batch_size=2,
+            random_eps=False,
+        )
+        x_train_adv_np = attack_np.generate(self.x_train_mnist)
+        x_test_adv_np = attack_np.generate(self.x_test_mnist)
+
+        attack_fw = ProjectedGradientDescent(
+            classifier,
+            eps=1,
+            eps_step=0.1,
+            max_iter=5,
+            norm=np.inf,
+            targeted=False,
+            num_random_init=0,
+            batch_size=2,
+            random_eps=False,
+        )
+        x_train_adv_fw = attack_fw.generate(self.x_train_mnist)
+        x_test_adv_fw = attack_fw.generate(self.x_test_mnist)
+
+        # Test
+        self.assertAlmostEqual(np.mean(x_train_adv_np), np.mean(x_train_adv_fw), places=6)
+        self.assertAlmostEqual(np.mean(x_test_adv_np), np.mean(x_test_adv_fw), places=6)
```

Comment:
How about testing the difference, that way we can make sure that data has been modified.
```suggestion
        self.assertAlmostEqual(np.mean(x_train_adv_np - self.x_train_mnist), np.mean(x_train_adv_fw - self.x_train_mnist), places=6)
        self.assertAlmostEqual(np.mean(x_test_adv_np - self.x_test_mnist), np.mean(x_test_adv_fw - self.x_test_mnist), places=6)
```